### PR TITLE
docs(core): Myriad of documentation changes including lots of new example code.

### PR DESCRIPTION
**Pull request by alxhub**

### CODE REVIEW

File: modules/angular2/examples/core/ts/bootstrap/bootstrap.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import {Component} from 'angular2/core';
+import {bootstrap} from 'angular2/platform/browser';
+
+// #docregion bootstrap
+@Component({selector: 'my-app', template: 'Hello {{ name }}!'})
+class MyApp {
+  name: string;
```
Comment (vicb) [inline modules/angular2/examples/core/ts/bootstrap/bootstrap.ts]: `name: string = 'World';` + remove ctor

File: modules/angular2/examples/testing/ts/testing.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  beforeEachProviders(() => [provide(MyService, {useClass: MyMockService})]);
+  it('uses MyService', inject([MyService], (service) => {
+                                               // service is an instance of MyMockService.
+                                           }));
+});
+// #enddocregion
+
+// #docregion afterEach
+describe('some component', () => {
+  afterEach((done) => { db.reset().then(() => done()); });
```
Comment (vicb) [inline modules/angular2/examples/testing/ts/testing.ts]: `then((_) => done())` for Dart ?

File: modules/angular2/src/testing/testing.ts [hunk 3]
```diff
@@ -15,27 +15,54 @@ export {expect, NgMatchers} from './matchers';
 var _global: jasmine.GlobalPolluter = <any>(typeof window === 'undefined' ? global : window);
 
 /**
- * See http://jasmine.github.io/
+ * Run a function (with an optional asynchronous callback) after each test case.
+ *
+ * See http://jasmine.github.io/ for more details.
+ *
+ * ## Example:
```
Comment (vicb) [inline modules/angular2/src/testing/testing.ts]: should we use `###` for Example (several occurences)

File: modules/angular2/examples/testing/ts/testing.ts (hunk 2 above)
Comment (alxhub) [inline modules/angular2/examples/testing/ts/testing.ts]: Done.

File: modules/angular2/src/testing/testing.ts (hunk 3 above)
Comment (alxhub) [inline modules/angular2/src/testing/testing.ts]: We use `##` for examples, per Naomi.

**Comment by vsavkin** [comment]
Merged via https://github.com/angular/angular/commit/778677ba75a562497e7dca7aea60496cf2c450f7

**Pull request closed without merging**